aopy/postproc.py

```diff
@@ -136,6 +136,109 @@ def get_trial_targets(trials, targets):
+
+    return inst_target_dir
+
+def mean_fr_inst_dir(data, cursorxpos, cursorypos, targetpos, data_binwidth, targetloc_binwidth, data_samplerate, cursor_samplerate):
+    '''
+    This function takes trial aligned neural data, cursor position, and target position and calculates
+    the mean firing rate per target location. 
+
+    Args:
+        data (ntime, nunit, ntrial): Trial aligned data
+        cursorxpos (ntime, ntrial): Trial aligned cursor position
+        cursorypos (ntime, ntrial): Trial aligned cursor position
+        targetpos (ntrial, 2): Target position for each trial. First column is x target position, second column is y target position.
+        data_binwidth (float): Bin size for neural data and cursor position. Can not be smaller than allowed by cursor position sampling rate.
+        targetloc_binwidth (float): Bin size for lumping target positions [deg]
+        data_samplerate (int): Sampling rate for data
+        cursor_samplerate (int): Sampling rate for cursor position
+
+    Returns:
+        (nunit, ndirection): Average firing rate per unit per direction bin.
+        
+    '''
+    # Check that binwidths are not lower than allowed by sampling rate
+    max_cursor_binwidth = 1/cursor_samplerate #[s/sample]
+    if data_binwidth < max_cursor_binwidth:
+        data_binwidth = max_cursor_binwidth
+    
+    ndatatime, nunit, ntrial = data.shape
+    ndirection = 360//targetloc_binwidth
+    nbins = math.ceil(ndatatime/(data_samplerate*data_binwidth)) # the number of bins
+
+    # Check target bin width is evenly spaced around task circle
+    if 360%targetloc_binwidth != 0:
+        warnings.warn("Target location bins are not evenly spaced. Please choose a bin width that evenly divides into 360 degrees")
+
+    # Bin neural data and cursor pos for each trial to make them the same samplingrate
+    binned_data = np.zeros((nbins, nunit, ntrial))*np.nan
+    for itrial in range(ntrial):
+        binned_data[:,:,itrial] = precondition.bin_spikes(data[:,:,itrial], data_samplerate, data_binwidth)
+
+    binned_cursorxpos = precondition.bin_spikes(cursorxpos, cursor_samplerate, data_binwidth)
+    binned_cursorypos = precondition.bin_spikes(cursorypos, cursor_samplerate, data_binwidth)
+
+    # Get instantaneous target location for each cursor pos (ntime, ntrial)
+    #print(binned_cursorxpos.shape, binned_cursorypos.shape, targetpos.shape)
+    inst_target_dir = get_inst_target_dir(binned_cursorxpos, binned_cursorypos, targetpos)
+    target_binid = 1 + (inst_target_dir-(np.deg2rad(targetloc_binwidth)/2))//np.deg2rad(targetloc_binwidth)
+    target_binid[target_binid==ndirection] = 0
+
+    # Average data and place into correct points in array
+    mean_dir_fr = np.zeros((nunit, ndirection))
+    for iunit in range(nunit):
+        for idir in range(ndirection):
+            temp_data = binned_data[:,iunit,:]
+            mean_dir_fr[iunit, idir] = np.mean(temp_data[target_binid==idir])
```

Comment:
have you tried something like this:
```suggestion
    for idir in range(ndirection):
        mean_dir_fr[:, idir] = np.mean(binned_data[target_binid==idir,:,:], axis=(0,2))
```